manim/renderer/cairo_renderer.py

```diff
@@ -179,30 +179,15 @@ def update_skipping_status(self):
         """
         if config["from_animation_number"]:
             if self.num_plays < config["from_animation_number"]:
-                config["skip_animations"] = True
+                self.skip_animations = True
         if config["upto_animation_number"]:
             if self.num_plays > config["upto_animation_number"]:
-                config["skip_animations"] = True
+                self.skip_animations = True
                 raise EndSceneEarlyException()
 
-    def revert_to_original_skipping_status(self):
-        """
-        Forces the scene to go back to its original skipping status,
-        by setting skip_animations to whatever it reads
-        from original_skipping_status.
-
-        Returns
-        -------
-        Scene
-            The Scene, with the original skipping status.
-        """
-        if hasattr(self, "original_skipping_status"):
-            config["skip_animations"] = self.original_skipping_status
-        return self
-
     def finish(self, scene):
-        config["skip_animations"] = False
+        self.skip_animations = self.original_skipping_status
         self.file_writer.finish()
         if config["save_last_frame"]:
-            self.update_frame(scene, ignore_skipping=False)
+            self.update_frame(scene, ignore_skipping=True)
```

Comment:
**Question** : Could you clarify the reason you put this to True ?

Reply:
My reasoning is that `skip_animations` is more of a local state variable. It's very ephemeral since it changes from frame to frame. But `config.save_last_frame` is more of a global state, it means that the user wants the last frame to be saved. So if the user wants the last frame to be saved, it makes sense to me that we should skip the internal/local/ephemeral state of `skip_animations`. So we should ignore it in this case. 

Once again, given the brittle state of the library I may have gotten the semantics wrong, but this reasoning makes sense to me :sweat_smile:

Reply:
i'm not sure to follow you. 
Did you mean from animation to animation instead of frame to frame ? 
Anyway, My reasonning is the following : 
it will likely cause issues. if `skip_animation` is set to true for any reason and `ignore_skipping` to False, it the animaton will be logically skipped. So, if we want to save the last frame, the frame will be empty as all the animations would have been skipped.

Reply:
You've convinced me. I just changed it.